src/main/java/org/reaktivity/nukleus/maven/plugin/internal/generate/VariantFlyweightGenerator.java

```diff
@@ -0,0 +1,1421 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.maven.plugin.internal.generate;
+
+
+import static com.squareup.javapoet.MethodSpec.constructorBuilder;
+import static com.squareup.javapoet.MethodSpec.methodBuilder;
+import static com.squareup.javapoet.TypeSpec.classBuilder;
+import static java.util.Collections.unmodifiableMap;
+import static javax.lang.model.element.Modifier.FINAL;
+import static javax.lang.model.element.Modifier.PRIVATE;
+import static javax.lang.model.element.Modifier.PUBLIC;
+import static javax.lang.model.element.Modifier.STATIC;
+import static org.reaktivity.nukleus.maven.plugin.internal.generate.TypeNames.BIT_UTIL_TYPE;
+import static org.reaktivity.nukleus.maven.plugin.internal.generate.TypeNames.DIRECT_BUFFER_TYPE;
+import static org.reaktivity.nukleus.maven.plugin.internal.generate.TypeNames.MUTABLE_DIRECT_BUFFER_TYPE;
+
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.CodeBlock;
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.FieldSpec.Builder;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+import com.squareup.javapoet.TypeName;
+import com.squareup.javapoet.TypeSpec;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeSet;
+
+public final class VariantFlyweightGenerator extends ClassSpecGenerator
+{
+    private final String baseName;
+    private final TypeSpec.Builder builder;
+    private final MemberFieldGenerator memberField;
+    private final KindConstantGenerator memberKindConstant;
+    private final MemberSizeConstantGenerator memberSizeConstant;
+    private final MemberOffsetConstantGenerator memberOffsetConstant;
+    private final MemberFieldValueConstantGenerator memberFieldValueConstant;
+    private final TryWrapMethodGenerator tryWrapMethod;
+    private final WrapMethodGenerator wrapMethod;
+    private final ToStringMethodGenerator toStringMethod;
+    private final KindAccessorGenerator kindAccessor;
+    private final MemberAccessorGenerator memberAccessor;
+    private final LimitMethodGenerator limitMethod;
+    private final BuilderClassGenerator builderClass;
+    private static final Map<String, String> NUMBER_BY_WORD;
+    private GetMethodGenerator getMethod;
+    private BitMaskConstantGenerator bitMaskConstant;
+
+    static
+    {
+        Map<String, String> numberByWord = new HashMap<>();
+        numberByWord.put("0", "zero");
+        numberByWord.put("1", "one");
+        NUMBER_BY_WORD = unmodifiableMap(numberByWord);
+    }
```

Comment:
Move `static` field definition and `static` initializer block to top of `class`.
`NUMBER_BY_WORD` is actually describing the opposite of lookup up words by numbers.
Suggest renaming to `NUMBER_WORDS` instead.